test/EndToEnd/ProjectBuildTests.cs

```diff
@@ -248,6 +248,29 @@ public void ItCanCreateItemTemplate(string templateName)
             Assert.True(directory.EnumerateFileSystemInfos().Any());
         }
 
+        [Theory]
+        // microsoft.dotnet.common.itemtemplates templates
+        [InlineData("class")]
+        [InlineData("struct")]
+        [InlineData("enum")]
+        [InlineData("record")]
+        [InlineData("interface")]
+        public void ItCanCreateItemTemplateWithProjectRestriction(string templateName, string language = "C#")
```

Comment:
consider adding at least one test for VB, and one test for default language (C#).